Makefile

```diff
@@ -9,21 +9,54 @@ DEFAULT_WORKSPACE_DIR = "./workspace"
 DEFAULT_MODEL = "gpt-4-0125-preview"
 CONFIG_FILE = config.toml
 
+UNAME_SYS = $(shell uname -s)
+# fallback to python3 if python is not available (e.g. on macOS)
+PYTHON_BIN = $(shell which python || which python3)
+ifeq ($(PYTHON_BIN),)
+	$(error "Python is not installed. Please install Python 3.")
+endif
+
+ifeq ($(shell which node), )
+	$(error "Node.js is not installed. Please install Node.js.")
+endif
+
+ifeq ($(shell which npm), )
+	$(error "npm is not installed. Please install npm.")
+endif
+
+INSTALL_PIPENV_BASE_CMD = $(PYTHON_BIN) -m pip install pipenv
+
+ifeq ($(shell which brew), )
+	INSTALL_PIPENV_BREW_CMD = echo "Error: Homebrew is not installed. Please install Homebrew from https://brew.sh/ and try again." && exit 1
+else
+	INSTALL_PIPENV_BREW_CMD = echo "Install pipenv from pip failed, try brew install pipenv..." && brew install pipenv
+endif
+
+ifeq ($(shell which pipenv), )
+	ifeq ($(UNAME_SYS),Darwin)
+		INSTALL_PIPENV_CMD = $(INSTALL_PIPENV_BASE_CMD) || $(INSTALL_PIPENV_BREW_CMD)
+	else
+		INSTALL_PIPENV_CMD = $(INSTALL_PIPENV_BASE_CMD)
+	endif
+else
+	INSTALL_PIPENV_CMD = echo "Pipenv is already installed."
+endif
+
 # Build
 build:
 	@echo "Building project..."
 	@echo "Pulling Docker image..."
 	@docker pull $(DOCKER_IMAGE)
 	@echo "Installing Python dependencies..."
-	@python -m pip install pipenv
-	@python -m pipenv install -v
+	@$(INSTALL_PIPENV_CMD)
+	@$(PYTHON_BIN) -m pipenv install
```

Comment:
With a brew installed pipenv on MacosX this gives me:

> $  python3 -m pipenv install
/usr/local/opt/python@3.12/bin/python3.12: No module named pipenv

`pipenv` is in the path and can be called directly.

Reply:
I could not use the brew version of python 3.12 myself I had to install from python.org